src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_UIElementCollection.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.UI.Xaml.Controls;
+using Private.Infrastructure;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls;
+
+[TestClass]
+public class Given_UIElementCollection
+{
+	[TestMethod]
+	[RunsOnUIThread]
+	[DataRow(2, 0)]
+	[DataRow(2, 2)]
+	[DataRow(2, 3)]
+	[DataRow(2, 4)]
+	public async Task When_Move_To_Valid(int from, int to)
```

Comment:
Can you please add test cases that test both moving an element backwards and forawrds in the children's list?
Especially forward, because when we remove the element, the target index is now -1.